### DISCUSSION TITLE
Possible to display image from rest response?

### DISCUSSION THREAD

**Discussion by apolitze** [Budibase/budibase#11936] [Help] [answered]
Hi Folks,
I'm new to bb and I have a challange in displaying the image value from rest call. I read some docs, but couldn't find a howto info, although I assume it might be some common case as I want to diplay generated qr code.
I configured a REST API datasource and I'm able to see data in response as shown:
![2023-09-30_12-58-55_Bd-Ds](https://github.com/Budibase/budibase/assets/31736279/94978798-0812-4cd1-a112-193c439da47a)

and I'm able to find a "value" on Design > image component, that is connected to repeater component (also tried with data provider component) and configured it in URL field(?) shown:
![2023-09-30_13-04-45_Bb-Design](https://github.com/Budibase/budibase/assets/31736279/0aa83b4d-4e7f-4f59-b398-488f453f53fa)

but the image is not displayed - I also tried the other repeater/data provider block options, but without success. How can I prepare "value" for URL field/image component or do I need to take a different approach .. any idea?
Thanks in advance

**Comment by ConorWebb96** [comment, marked as answer]
Hey @apolitze,

Is there a chance you could change the output of that rest API to a base64 image string. This should allow you to use the image component to display the QR code.

This is a very strange URL? I'm not seeing any http protocol/address to were the image is stored. Additionally you seem to have encoding mismatches judging off the `?` symbols contained within your JSON response? 

Normally a URL for an image would look something like this; `https://placehold.co/600x400` (This is a very simple example.) Locally stored images via the attachment in BB internal db would look something like this `/files/signed/prod-budi-app-assets/app_2341290865202/attachments/fe1236321.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=budibase%2F20231002%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20231002T071621Z&X-Amz-Expires=3600&X-Amz-Signature=3241246&X-Amz-SignedHeaders=host`.

I hope this helps.

**Reply by apolitze** [reply]
Hi @ConorWebb96,

thank you for your comment and infos! I changed the source to get a base64 response, which is currently as shown:
![2023-10-02_16-55_Bb_data](https://github.com/Budibase/budibase/assets/31736279/f2dc965b-35f9-45df-96f1-970f74b04b97)

the design "sees" the data as string, but does not display the qr code yet:
![2023-10-02_16-55_Bb_design](https://github.com/Budibase/budibase/assets/31736279/d8acca0c-85af-4eca-a26a-727d0f9b2153)

I'm still investigating..

**Comment by ConorWebb96** [comment]
Hey @apolitze,

This definitely works as i've just copied my base64 converted image and pasted the code into the bindable image area and it has displayed the image.

I think you just aren't using the correct bindable to retrieve the data. I'd recommend using the paragraph component to see what each stage of the binding returns, that way you can break it down step by step until you retrieve the base64 code needed for the image.

<img width="869" alt="Screenshot 2023-10-03 at 07 33 40" src="https://github.com/Budibase/budibase/assets/126772285/262623ce-4d89-47b5-8c69-59966a868fad">

I hope this helps!

**Reply by apolitze** [reply]
Hi @ConorWebb96,
Yes, for me the copied response works fine, too, and you're right the binding is wrong. As there is a transformation into an array - with latest try with data provider - I just found out:
`{{ New Data Provider.Rows.[0].value }}` works

Thanks a lot @ConorWebb96 !
